File: solution/2000-2099/2016.Maximum Difference Between Increasing Elements/Solution.go
func maximumDifference(nums []int) int {
	mi, ans := nums[0], -1
	for i, n := 1, len(nums); i < n; i++ {
		if nums[i] > mi {
			ans = max(ans, nums[i]-mi)
		} else {
			mi = nums[i]
		}
	}
	return ans
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: solution/2000-2099/2016.Maximum Difference Between Increasing Elements/Solution.rs
impl Solution {
    pub fn maximum_difference(nums: Vec<i32>) -> i32 {
        let mut mi = i32::MAX;
        let mut ans = -1;

        for &x in &nums {
            if x > mi {
                ans = ans.max(x - mi);
            } else {
                mi = x;
            }
        }

        ans
    }
}
